cunumeric/random/bitgenerator.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from cunumeric.array import ndarray
+from cunumeric.config import BitGeneratorType
+from cunumeric.runtime import runtime
+
+
+class BitGenerator:
+    def __init__(self, seed=None, generatorType=BitGeneratorType.DEFAULT):
+        if type(self) is BitGenerator:
+            raise NotImplementedError(
+                "BitGenerator is a base class and cannot be instantized"
+            )
+        self.handle = runtime.bitgenerator_create(generatorType)
+        if seed is not None:
+            runtime.bitgenerator_set_seed(self.handle, seed)
+
+    def __del__(self):
+        runtime.bitgenerator_destroy(self.handle)
```

Comment:
My previous comment seems to have been removed for some reason, but it's not safe to do something that leads to a task launch in a Python destructor. Please move this outside the destructor. You can simply record the destruction of a bit generator in the destructor and later batch destroy them whenever you get a chance. For example, when you create a new generator, you can ask the runtime to destroy all outstanding destruction requests for bit generators.

Reply:
I maintained an explicit destroy task and added a lazy destroy that is then performed on the next create/destroy task